File: DataStructures/Stacks/Stack.rb
class Node
  attr_accessor :data, :previous

  def initialize(data)
    @data = data
    @previous = nil
  end
end

class Stack
  attr_accessor :length, :top

  def initalize()
    @top = nil
    @length = 0
  end

  def push(data)
    node = Node.new(data)
    node.previous = @top
    @top = node
    @length =+ 1

    return @top
  end

  def pop()
    current = @top

    @top = @top.previous
    @length =- 1
    return current.data
  end
end

s = Stack.new()
s.push(2)
s.push(12)

print s.pop()

File: DataStructures/Stacks/Stack.py
class Node(object):
    def __init__(self, data = None, previous = None):
        self.data = data
        self.previous = previous

class Stack(object):
    def __init__(self):
        self.top = None
        self.size = 0

    def push(self, data):
        node = Node(data)
        node.previous = self.top
        self.top = node
        self.size += 1

        return self.top

    def pop(self):
        current = self.top

        self.top = self.top.previous
        self.size -= 1

        return current.data
